bigquery/bigquery_storage_quickstart/main.go

```diff
@@ -224,10 +227,31 @@ func processStream(ctx context.Context, client *bqStorage.BigQueryReadClient, st
 			if err == io.EOF {
 				return nil
 			}
-			if err != nil {
-				retries++
-				if retries >= retryLimit {
-					return fmt.Errorf("processStream retries exhausted: %v", err)
+			if err == nil {
+				// Reset retries after a successful response.
+				retries = 0
+			} else {
+				// If there is an error, check whether it is a retryable error with a retry delay and sleep instead of increasing retries count.
+				var retryDelayDuration time.Duration
+				if errorStatus, ok := status.FromError(err); ok && errorStatus.Code() == codes.ResourceExhausted {
```

Comment:
Do we ever anticipate the backend adding RetryInfo to responses other than ResourceExhausted?  If so, perhaps we don't special case the `ResourceExhausted`.

Reply:
BigQuery storage backend will only return a RetryInfo with ResourceExhausted.